### PR TITLE
[OpsGenie v3] Get Team Routing Rules

### PR DESCRIPTION
**Pull request by ennozdd**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml [hunk 1]
```diff
@@ -1259,7 +1262,14 @@ script:
       description: Whether the request was successful.
       type: Boolean
 
-  dockerimage: demisto/python3:3.10.12.63474
+  - name: opsgenie-get-team-routing-rules
```
Comment (mmhw) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1265]: Please add the `outputs` of the command.

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml [hunk 2]
```diff
@@ -116,6 +116,9 @@ script:
       name: source
     - description: Additional alert note.
       name: note
+    - name: details
+      description: Comma-separated key-value pairs to use as custom properties of the alert. JSON format is also supported when used within an automation.
```
Comment (mmhw) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:120]:
```suggestion
      description: Comma-separated key=value pairs to use as custom properties of the alert. JSON format is also supported when used within an automation.
```
Maybe you should also add an example

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py [hunk 3]
```diff
@@ -867,6 +876,16 @@ def get_teams(client: Client, args: Dict[str, Any]) -> CommandResults:
...
 
 
+def get_team_routing_rules(client: Client, args: Dict[str, Any]) -> CommandResults:
+    result = client.get_team_routing_rules(args)
+    return CommandResults(
+        outputs_prefix="OpsGenie.Team.RoutingRules",
+        outputs=result.get("data"),
+        readable_output=tableToMarkdown("OpsGenie Team Routing Rules", result.get("data")),
+        raw_response=result
+    )
```
Comment (mmhw) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:886]:
```suggestion
    data = result.get("data")
    return CommandResults(
        outputs_prefix="OpsGenie.Team.RoutingRules",
        outputs=data,
        readable_output=tableToMarkdown("OpsGenie Team Routing Rules", data),
        raw_response=result
    )
```

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py [hunk 4]
```diff
@@ -77,6 +77,10 @@ def responders_to_json(responders: List, responder_key: str, one_is_dict: bool =
 
     def create_alert(self, args: dict):
         args['responders'] = argToList(args.get('responders'))
+        if not isinstance(args.get('details'), dict):
+            args['details'] = {key_value.split('=')[0]:key_value.split('=')[1]
+                               for key_value in args.get('details').split(',')}
```
Comment (mmhw) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py]:
```suggestion
        if args.get('details') and not isinstance(args.get('details'), dict):
            args['details'] = {key_value.split('=')[0]: key_value.split('=')[1]
                               for key_value in argToList(args.get('details'))}
```
There should be something like that, `details` is not a mandatory argument

**Review by mmhw — COMMENTED** [review]
Hi @ennozdd,
Thank you for your contribution!
Good work :)

Please note my comments

**Comment by ennozdd** [comment]
@mmhw  done :)

**Comment by mmhw** [comment]
Hi @ShirleyDenkberg,
Could you please look at this PR?
Thanks!

File: Packs/OpsGenie/Integrations/OpsGenieV3/README.md [hunk 5]
```diff
@@ -1839,3 +1840,31 @@ Removed the following commands in this version:
...
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| OpsGenie.Team.RoutingRules.name | unknown | Name of the routing rule. | 
+| OpsGenie.Team.RoutingRules.order | unknown | Order of the routing rule. | 
+| OpsGenie.Team.RoutingRules.id | unknown | ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.timezone | unknown | Timezone of the routing rule. | 
+| OpsGenie.Team.RoutingRules.teamId | unknown | Team ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.customerId | unknown | Customer ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.notify.id | unknown | Notify ID of the routing rule |
```
Comment (ShirleyDenkberg) [inline Packs/OpsGenie/Integrations/OpsGenieV3/README.md:1868]:
```suggestion
| OpsGenie.Team.RoutingRules.notify.id | unknown | Notify ID of the routing rule. | 
```

File: Packs/OpsGenie/Integrations/OpsGenieV3/README.md [hunk 6]
```diff
@@ -1839,3 +1840,31 @@ Removed the following commands in this version:
...
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| OpsGenie.Team.RoutingRules.name | unknown | Name of the routing rule. | 
+| OpsGenie.Team.RoutingRules.order | unknown | Order of the routing rule. | 
+| OpsGenie.Team.RoutingRules.id | unknown | ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.timezone | unknown | Timezone of the routing rule. | 
+| OpsGenie.Team.RoutingRules.teamId | unknown | Team ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.customerId | unknown | Customer ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.notify.id | unknown | Notify ID of the routing rule | 
+| OpsGenie.Team.RoutingRules.notify.name | unknown | Notify Name of the routing rule |
```
Comment (ShirleyDenkberg) [inline Packs/OpsGenie/Integrations/OpsGenieV3/README.md:1869]:
```suggestion
| OpsGenie.Team.RoutingRules.notify.name | unknown | Notify name of the routing rule. | 
```

File: Packs/OpsGenie/Integrations/OpsGenieV3/README.md [hunk 7]
```diff
@@ -1839,3 +1840,31 @@ Removed the following commands in this version:
...
+| --- | --- | --- |
+| OpsGenie.Team.RoutingRules.name | unknown | Name of the routing rule. | 
+| OpsGenie.Team.RoutingRules.order | unknown | Order of the routing rule. | 
+| OpsGenie.Team.RoutingRules.id | unknown | ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.timezone | unknown | Timezone of the routing rule. | 
+| OpsGenie.Team.RoutingRules.teamId | unknown | Team ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.customerId | unknown | Customer ID of the routing rule. | 
+| OpsGenie.Team.RoutingRules.notify.id | unknown | Notify ID of the routing rule | 
+| OpsGenie.Team.RoutingRules.notify.name | unknown | Notify Name of the routing rule | 
+| OpsGenie.Team.RoutingRules.notify.type | unknown | Notify Type of the routing rule |
```
Comment (ShirleyDenkberg) [inline Packs/OpsGenie/Integrations/OpsGenieV3/README.md:1870]:
```suggestion
| OpsGenie.Team.RoutingRules.notify.type | unknown | Notify type of the routing rule. | 
```

File: Packs/OpsGenie/ReleaseNotes/2_0_17.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#### Integrations
+##### OpsGenie v3
+
+- Added a new command **opsgenie-get-team-routing-rules**.
+- Added a new argument **details** to the command **opsgenie-create-alert**.
```
Comment (ShirleyDenkberg) [inline Packs/OpsGenie/ReleaseNotes/2_0_17.md:5]:
```suggestion
- Added a new argument **details** to the command ***opsgenie-create-alert***.
```

File: Packs/OpsGenie/ReleaseNotes/2_0_17.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#### Integrations
+##### OpsGenie v3
+
+- Added a new command **opsgenie-get-team-routing-rules**.
```
Comment (ShirleyDenkberg) [inline Packs/OpsGenie/ReleaseNotes/2_0_17.md:4]:
```suggestion
- Added a new command ***opsgenie-get-team-routing-rules***.
```

**Comment by ShirleyDenkberg** [comment]
@mmhw @GuyAfik Doc review completed.

**Comment by ennozdd** [comment]
@mmhw @ShirleyDenkberg done

**Comment by mmhw** [comment]
Hi @ennozdd,
The PR is ready.
Only note that i changed the `output` prefix from "OpsGenie.Team.RoutingRules" to "OpsGenie.TeamRoutingRule", so if you can confirm everything is working as expected and we will merge it.

Thanks!

**Comment by ennozdd** [comment]
@mmhw all good, thank you!

**Review by mmhw — APPROVED** [review]
Very Good!

**Pull request merged**